support/node-production-aws-sns/index.js

```diff
@@ -0,0 +1,65 @@
+const { ApiPromise, WsProvider } = require('@polkadot/api');
+const AWS = require('aws-sdk');
+const Settings = require('./settings.json')
+
+// Create a promise API instance of the passed in node address.
+async function createPromiseApi(nodeAddress) {
+    const provider = new WsProvider(nodeAddress);
+    const api = await new ApiPromise({ provider });
+    await api.isReady;
+    return api;
+}
+
+async function main() {
+    const api = await createPromiseApi(Settings.nodeAddress);
+
+    AWS.config.update({region: Settings.region});
+    const sns = new AWS.SNS({apiVersion: Settings.awsApiVersion});
+    const topics = await sns.listTopics().promise();
+    const topicArn = topics.Topics.find(t => t.TopicArn.includes(Settings.topicName)).TopicArn;
+
+    let lastHeaderHex = Settings.lastHeaderHex;
+    let lastNewBlockAt = Date.now();
+    let sendingSns = false;
+    let sleepTime = 0;
+    while (true) {
+        await new Promise(r => setTimeout(r, sleepTime));
+        try {
+            sleepTime = Settings.sleepIntervalMs
+            const signedBlock = await api.rpc.chain.getBlock();
+            const currentHeader = signedBlock.block.header.hash;
+
+            if (currentHeader.toHex() != lastHeaderHex) {
+                lastHeaderHex = currentHeader.toHex();
+                lastNewBlockAt = Date.now();
+                console.log(`Block header hex: ${lastHeaderHex} at ${new Date().toISOString()}`);
+                continue;
+            }
+
+            console.log(new Date().toISOString(), `Have not seen new block for ${new Date() - lastNewBlockAt}ms`);
+
+            if (sendingSns) {
+                continue;
+            }
+
+            if (Date.now() - lastNewBlockAt > Settings.requireNewBlockEveryMs) {
+                sleepTime = 1000 * 60 * Settings.minutesUntilNextCheckAfterAlarm;
```

Comment:
How about moving this to the end of the block and sleeping the amount directly. Since this is the only place we change sleepTime, then we wouldn't need to have a local variable and can have the initial sleep always do what's configured.